src/qcodes/parameters/parameter_base.py

```diff
@@ -778,6 +780,14 @@ def set_wrapper(value: ParamDataType, **kwargs: Any) -> None:
 
                     self.cache._update_with(value=val_step, raw_value=raw_val_step)
 
+                    if ParameterBase._value_changed_callback is not None:
```

Comment:
Is there a reason for not doing 
```suggestion
                    if self._value_changed_callback is not None:
```

which would allow a custom calllback in subclasses?

Reply:
I made my comments in my previous PR (which I emailed you about accidentally closing it). (including naming suggestion/convention) Link: https://github.com/microsoft/Qcodes/pull/6881

So my comment there is as follows:
We had to change our approach to tracking parameter changes. Instead of monitoring each individual instance, we now track changes on a global level. This change came after many discussions with our physicists and engineers. Our systems rely heavily on third-party instruments and drivers, and the old method wasn't practical. It would have required updating thousands of parameter instances across different instruments and submodules - many of which we don't control because we use their drivers.

With global tracking, we can easily keep track of all parameter changes in our system, including third-party instruments, without having to change each one individually.

Reply:
I have just made the changes based on your suggestions, may I ask how it is now?

Example usage:

```python
def my_callback(param: ParameterBase, value: Any) -> None:
    print(f"Parameter {param.name} changed to {value}")

# Set global callback
ParameterBase.set_global_value_changed_callback(my_callback)

# Clear global callback
ParameterBase.set_global_value_changed_callback(None)
```

Reply:
Thanks that looks good I think